contrib/apache-httpd/mod_arpa2_tlspool.c

```diff
@@ -0,0 +1,314 @@
+                break;
+            case SSL_CVERIFY_OPTIONAL:
+                tlsdata_now.flags |= PIOF_STARTTLS_REQUEST_REMOTEID;
+                break;
+            default:
+                tlsdata_now.flags |= PIOF_STARTTLS_IGNORE_REMOTEID;
+                break;
+        }
+        note = apr_psprintf(c->pool, "tlspool_pre_connection: c = %pp, pool = %pp, old = %d, nVerifyCLient = %d, flags = 0x%08x",
+                        (void*) c, (void*) c->pool, cnx, pMainConfig->nVerifyClient, tlsdata_now.flags);
+        trace_nocontext(c->pool, __FILE__, __LINE__, note);
+        if (-1 == tlspool_starttls (cnx, &tlsdata_now, &privdata, namedconnect_vhost)) {
+            note = apr_psprintf(c->pool, "Failed to STARTTLS on Apache: errno = %d", errno);
+            trace_nocontext(c->pool, __FILE__, __LINE__, note);
+            if (privdata.plainfd >= 0) {
+                close (privdata.plainfd);
+            }
+            exit (1);
+        }
+        os_sock = (apr_os_sock_t) privdata.plainfd;
+        apr_os_sock_put(&apr_socket, &os_sock, c->pool);
+
+        /*
+         * Log the call and exit.
+         */
+        note = apr_psprintf(c->pool, "tlspool_pre_connection: new = %d, localid = %s", privdata.plainfd, tmp);
+        trace_nocontext(c->pool, __FILE__, __LINE__, note);
+    } else {
+        trace_nocontext(c->pool, __FILE__, __LINE__, "tlspool_pre_connection: TLSPoolEnable off");
+    }
+    return OK;
+}
+
+static const char *ssl_cmd_verify_parse(cmd_parms *parms,
+                                        const char *arg,
+                                        ssl_verify_t *id)
+{
+    if (strcEQ(arg, "none") || strcEQ(arg, "off")) {
+        *id = SSL_CVERIFY_NONE;
+    }
+    else if (strcEQ(arg, "optional")) {
+        *id = SSL_CVERIFY_OPTIONAL;
+    }
+    else if (strcEQ(arg, "require") || strcEQ(arg, "on")) {
+        *id = SSL_CVERIFY_REQUIRE;
+    }
+    else if (strcEQ(arg, "optional_no_ca")) {
+        *id = SSL_CVERIFY_OPTIONAL_NO_CA;
+    }
+    else {
+        return apr_pstrcat(parms->temp_pool, parms->cmd->name,
+                           ": Invalid argument '", arg, "'",
+                           NULL);
+    }
+
+    return NULL;
+}
+
+const char *ssl_cmd_SSLVerifyClient(cmd_parms *cmd,
+                                    void *dcfg,
+                                    const char *arg)
+{
+    tlspool_server_config *pConfig = our_sconfig(cmd->server);
+
+    ssl_verify_t mode = SSL_CVERIFY_NONE;
+    const char *err;
+
+    if ((err = ssl_cmd_verify_parse(cmd, arg, &mode))) {
+        return err;
+    }
+
+    server_rec* server = cmd->server;
+    char* server_hostname = server->server_hostname;
+    char* note = apr_psprintf(cmd->temp_pool, "ssl_cmd_SSLVerifyClient: server_hostname = %s", server_hostname);
+    trace_nocontext(cmd->temp_pool, __FILE__, __LINE__, note);
+
+    pConfig->nVerifyClient = mode;
+    return NULL;
+}
+
+static const command_rec tlspool_cmds[] =
```

Comment:
This table is far from complete! Please check with the document for the others and supply them along with this?